File: tests/26_sra_1.go
package main

import . "glyph/tests/interp"

func t() {
    c := []uint64{
    }
    i := []uint16{
        CPU_encode_op_movi_i64(0,-8),
        CPU_encode_op_movi_i64(1,2),
        CPU_encode_op_sra_i64(2,0,1),
        CPU_encode_op_break(0),
    }
    CPU_test("sra_1", c, i);
}

func main() {
    t()
}

File: tests/26_sra_1.py
#!/usr/bin/env python3

from interp import *

def t():
    c = [
    ]
    i = [
        cpu_encode_op_movi_i64(0,-8),
        cpu_encode_op_movi_i64(1,2),
        cpu_encode_op_sra_i64(2,0,1),
        cpu_encode_op_break(0),
    ]
    cpu_test("sra_1", c, i)

t()
